lib/matplotlib/text.py

```diff
@@ -2092,7 +2093,9 @@ def draw(self, renderer):
                 self.arrow_patch.figure = self.figure
             self.arrow_patch.draw(renderer)
 
-        Text.draw(self, renderer)
+        astext = copy.copy(self)
+        astext.__class__ = Text
```

Comment:
Yes, I agree, while going through the Text module, Annotation does seem like it might be better suited for composition rather than inheritance. There is some precedence for modifying the __class__ attribute in the codebase currently but ultimately it is a hack to get the desired MRO.

One alternative approach is to refactor the `Text.draw` method to not reuse `bbox_artist` but that just seems like a step in the wrong direction by duplicating code. Another is to design a `Text` initialization that takes the derived class and returns a `Text` instance.